src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,363 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to and navigating through them.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub mod project_name;
+
+pub use breadcrumb::Breadcrumb;
+pub use project_name::ProjectName;
+
+use crate::graph_editor::MethodPointer;
+
+use enso_frp as frp;
+use enso_protocol::language_server::ExpressionId;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::text::text_field::FocusManager;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use std::cmp;
+
+
+
+// =================
+// === LocalCall ===
+// =================
+
+/// A specific function call occurring within another function's definition body.
+///
+/// This is a single item in ExecutionContext stack.
+#[derive(Clone,Debug,Eq,PartialEq)]
+pub struct LocalCall {
+    /// An expression being a call to a method.
+    pub call       : ExpressionId,
+    /// A pointer to the called method.
+    pub definition : MethodPointer,
+}
+
+
+
+// =================
+// === Constants ===
+// =================
+
+// FIXME[dg] hardcoded literal for glyph of height 12.0. Copied from port.rs
+const GLYPH_WIDTH       : f32 = 7.224_609_4;
+const VERTICAL_MARGIN   : f32 = GLYPH_WIDTH;
+const HORIZONTAL_MARGIN : f32 = GLYPH_WIDTH;
+const TEXT_SIZE         : f32 = 12.0;
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Push breadcrumb.
+    pub push_breadcrumb : frp::Source<Option<LocalCall>>,
+    /// Pop breadcrumb.
+    pub pop_breadcrumb : frp::Source,
```

Comment:
if we don't align multiple lines together, the `:` should not have spaces or any side